### PR TITLE
Expose WCSCOMPARE_* constants through the wcs module

### LINKED ISSUE #11631: Where are WCS comparison constants defined?
[`astropy.wcs.Wcsprm.compare`](https://docs.astropy.org/en/stable/api/astropy.wcs.Wcsprm.html?highlight=wcscompare_ancillary#astropy.wcs.Wcsprm.compare) documents a number of constants (e.g. `WCSCOMPARE_ANCILLARY`) that can be used with the `cmp` argument to modify how the comparison between two WCSs is done. It's not clear to me where these are defined however.

I've tried under `astropy.wcs` and `astropy.wcs.Wcsprm`, but maybe I'm missing somewhere? Wherever they are defined should be documented in the docstring of `.compare()`.

### PR DESCRIPTION
**Pull request by mcara**
This PR exposes `WCSCOMPARE_*` constants from WCSLIB through the `astropy.wcs` module.

Given that these constants were defined in WCSLIB since, at least 2014 or maybe even since the beginning, I consider this to be a bug since documentation refers to these constants.

Fixes #11631

### CODE REVIEW

**Comment by pllim** [comment]
If it is to fix something that been there since 2014, why not 4.0.6? But given no one asked for it till now, maybe there is no need to backport and just put it in 4.3. 🤷  I'll let `wcs` maintainers decide.

**Review by pllim — APPROVED** [review]
LGTM and tests passed. But I'll keep it open in case others want to review.

**Review by nden — APPROVED** [review]

**Pull request merged**
